File: GXpert/GXpert.Web/Modules/ServerTypes/Exams/ExamColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EExamTypes } from "../Web/Enums.EExamTypes";
import { ExamRow } from "./ExamRow";

export interface ExamColumns {
    Id: Column<ExamRow>;
    Title: Column<ExamRow>;
    EExamType: Column<ExamRow>;
    EExamState: Column<ExamRow>;
    QuestionPaperPath: Column<ExamRow>;
    ModelAnswerPaperPath: Column<ExamRow>;
    ExamDuration: Column<ExamRow>;
    MaxMarks: Column<ExamRow>;
    NegativeMarks: Column<ExamRow>;
    EExamNavigation: Column<ExamRow>;
    ESectionNavigation: Column<ExamRow>;
    EQuestionNavigation: Column<ExamRow>;
    EResultType: Column<ExamRow>;
    EOptionDisplayType: Column<ExamRow>;
    HasNegativeMarketing: Column<ExamRow>;
    Instructions: Column<ExamRow>;
    SearchTags: Column<ExamRow>;
}

export class ExamColumns extends ColumnsBase<ExamRow> {
    static readonly columnsKey = 'Exams.Exam';
    static readonly Fields = fieldsProxy<ExamColumns>();
}

[EExamTypes]; // referenced types

File: GXpert/GXpert.Web/Modules/Exams/Exam/ExamColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Exams.Columns;

[ColumnsScript("Exams.Exam")]
[BasedOnRow(typeof(ExamRow), CheckNames = true)]
public class ExamColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public short EExamType { get; set; }
    public short EExamState { get; set; }
    public string QuestionPaperPath { get; set; }
    public string ModelAnswerPaperPath { get; set; }
    public int ExamDuration { get; set; }
    public float MaxMarks { get; set; }
    public float NegativeMarks { get; set; }
    public short EExamNavigation { get; set; }
    public short ESectionNavigation { get; set; }
    public short EQuestionNavigation { get; set; }
    public short EResultType { get; set; }
    public short EOptionDisplayType { get; set; }
    public bool HasNegativeMarketing { get; set; }
    public string Instructions { get; set; }
    public string SearchTags { get; set; }
}